src/main/java/builder/TableProduct.java

```diff
@@ -22,29 +24,49 @@ public Long getUniqueRowsCount() {
     }
 
     public void setUniqueRowsCount(Long uniqueRowsCount) {
-        this.uniqueRowsCount = uniqueRowsCount;
+        if (uniqueRowsCount >= 0) {this.uniqueRowsCount = uniqueRowsCount;}
+        else {throw new IllegalArgumentException("uniqueRowsCount less or equals 0");}
     }
 
     public Long getRowsCount() {
         return rowsCount;
     }
 
     public void setRowsCount(Long rowsCount) {
-        this.rowsCount = rowsCount;
+        if(rowsCount > 0) {this.rowsCount = rowsCount;}
+        else {throw new IllegalArgumentException("rowsCount less or equals 0");}
+
     }
 
     public Boolean hasDuplicates() {
         return (rowsCount == uniqueRowsCount);
     }
 
-    public void addMetric(Metrics obj){listOfMetrics.add(obj);}
+    public void addMetric(Metrics obj){
+        try {
+            if (obj == null){
+                throw new NullPointerException("Bad object add");
+            } else {
+                listOfMetrics.add(obj);
+            }
+        } catch(NullPointerException e)  {
+            e.printStackTrace();
+            return;
+        }
+    }
+
     public void setDate() {calendar = Calendar.getInstance();}
 
     public String getDate() {
-        return new SimpleDateFormat("yyyyMMdd HH:mm:ss").format(calendar.getTimeInMillis());
+        try {
+            return new SimpleDateFormat("yyyyMMdd HH:mm:ss").format(calendar.getTimeInMillis());
+        } catch (NullPointerException e) {
+            throw new IllegalArgumentException("Date did not set");

```

Comment:
Please use this constructor to retain the original Exception:
`public Exception(String message, Throwable cause)`